src/packet/sym_encrypted_protected_data.rs

```diff
@@ -67,12 +67,120 @@ impl SymEncryptedProtectedData {
         })
     }
 
-    /// Same as [`encrypt_with_rng`], but uses [`thread_rng`] for RNG.
+    /// Encrypts the data using the given symmetric key.
+    pub fn encrypt_with_rng_seipdv2<R: CryptoRng + Rng>(
+        rng: &mut R,
+        sym_alg: SymmetricKeyAlgorithm,
+        aead: AeadAlgorithm,
+        chunk_size: u8,
+        session_key: &[u8],
+        plaintext: &[u8],
+    ) -> Result<Self> {
+        // FIXME: DRY decrypt/encrypt code duplication
+
+        // Initial key material is the session key.
+        let ikm = session_key;
+
+        // Generate new salt for this seipd packet.
+        let mut salt = [0u8; 32];
+        rng.fill(&mut salt[..]);
+
+        let info = [
+            Tag::SymEncryptedProtectedData.encode(), // packet type
+            0x02,                                    // version
+            sym_alg.into(),
+            aead.into(),
+            chunk_size,
+        ];
+
+        let chunk_size_expanded = expand_chunk_size(chunk_size);
+        let hk = hkdf::Hkdf::<Sha256>::new(Some(&salt[..]), ikm);
+        let mut okm = [0u8; 42];
+        hk.expand(&info, &mut okm).expect("42");
+        debug!("info: {} - hkdf: {}", hex::encode(info), hex::encode(okm));
+        let message_key = &okm[..sym_alg.key_size()];
+        let raw_iv_len = aead.nonce_size() - 8;
+        let iv = &okm[sym_alg.key_size()..sym_alg.key_size() + raw_iv_len];
+        let mut nonce = vec![0u8; aead.nonce_size()];
+        nonce[..raw_iv_len].copy_from_slice(iv);
+
+        debug!("message_key: {}", hex::encode(message_key));
+        debug!("iv: {}", hex::encode(iv));
+        debug!("nonce: {}", hex::encode(&nonce));
```

Comment:
note: before merging please make a pass to not log any sensitive material at any log level

Reply:
right. i think 62ab81dd17bb79fdb79c8dc760e6c6b20560b3e9 drops all of them.